### PR TITLE
Hw 1

### PR DESCRIPTION
**Pull request by AlenaKharytonchyk**
Created:
⚛️ Counter

⚛️ SearchForm

⚛️ GenreSelect

 according to the home task

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,32 @@
-import logo from './logo.svg';
 import './App.css';
+import Title from "./components/title/Title";
```
Comment (AliaksandrZahorski) [inline src/App.js]: it is better keep simple path, e.g. import { Title, Counter, ... } from "./components";

File: src/components/Counter.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import Title from "./title/Title";
+export default class Counter extends React.Component {
+    constructor({initialValue}) {
+        super();
+        this.state={
+            initialValue : initialValue
+        }
+    }
+    counterUpdate (priority) {
+        this.setState({initialValue: priority ? this.state.initialValue + 1 : this.state.initialValue - 1 })
```
Comment (AliaksandrZahorski) [inline src/components/Counter.js]: a kind of anti pattern, better use setState((prevState) => prevState + 1);

**Review by AliaksandrZahorski — CHANGES_REQUESTED** [review]

File: src/components/Counter.js (hunk 2 above)
Comment (AlenaKharytonchyk) [reply src/components/Counter.js]: Done

File: src/App.js (hunk 1 above)
Comment (AlenaKharytonchyk) [reply src/App.js]: Done

**Comment by AliaksandrZahorski** [comment]
approved

**Pull request merged**